akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala

```diff
@@ -111,21 +124,32 @@ trait PersistentFSM[S <: FSMState, D, E] extends PersistentActor with Persistent
       //Persist the events and apply the new state after all event handlers were executed
       var nextData: D = stateData
       var handlersExecutedCounter = 0
+      var doSnapshot: Boolean = false
+      val moduloCondition: Long ⇒ Boolean = snapshotAfter match {
```

Comment:
I think this function can be defined as an instance `val` instead and then it doesn't have to check the `snapshotAfter` each time, but that would have the same bin compat problem as the other field that is added.